### PR TITLE
Implement draggable annotations on MapView. Closes #2512

### LINKED ISSUE #2512: [MapView] Draggable annotations
For our app it would be nice if the user has the ability to drag a marker on the map. Currently, it doesn't [look supported](https://facebook.github.io/react-native/docs/mapview.html#annotations), but I imagine an implementation would have the properties ```draggable``` and ```onDrop```. If it sounds interesting to add, I can work on this feature and create a PR.

### PR DESCRIPTION
**Pull request by browniefed**

### CODE REVIEW

File: Libraries/Components/MapView/MapView.js [hunk 1]
```diff
@@ -35,6 +35,14 @@ type MapRegion = {
   longitudeDelta: number;
 };
 
+var ANNOTATION_DRAG_STATE = [
+  'Idle',
+  'Starting',
+  'Dragging',
+  'Canceling',
```
Comment (nicklockwood) [inline Libraries/Components/MapView/MapView.js]: The preferred way to handle enums like this in RN is with an RCTConvert category that maps lower-case string names to the underlying native enum automatically, and a flow $Enum<> in the JS.

See AlertIOS/RCTAlertManager for an example.

File: Libraries/Components/MapView/MapView.js [hunk 2]
```diff
@@ -196,6 +204,10 @@ var MapView = React.createClass({
        * Whether the pin drop should be animated or not
        */
       animateDrop: React.PropTypes.bool,
+      /**
+       * Whether the pin should be draggable or not
+       */
+      draggable: React.PropTypes.bool,
```
Comment (nicklockwood) [inline Libraries/Components/MapView/MapView.js]: @platform ios

File: Libraries/Components/MapView/MapView.js [hunk 3]
```diff
@@ -299,8 +311,16 @@ var MapView = React.createClass({
...
      */
     active: React.PropTypes.bool,
+
+    /**
+     * Callback that is called when the state of of annotations drag changes
+     */
+    onAnnotationDragChange: React.PropTypes.func,
   },
 
+
```
Comment (nicklockwood) [inline Libraries/Components/MapView/MapView.js]: Please remove this whitespace

File: Libraries/Components/MapView/MapView.js [hunk 4]
```diff
@@ -196,6 +204,10 @@ var MapView = React.createClass({
        * Whether the pin drop should be animated or not
        */
       animateDrop: React.PropTypes.bool,
```
Comment (nicklockwood) [inline Libraries/Components/MapView/MapView.js]: Blank line between props

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
@browniefed Hey, can you rebase it against master?

**Comment by christopherdro** [comment]
Closes #2512

**Comment by browniefed** [comment]
Will rebase tomorrow, didn't see the message 8 days ago

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by nicklockwood** [comment]
This was actually broken in it's current form - the onAnnotationDragChange property wasn't exported, and the draggable property had no effect.

Not sure if it was a merge issue or what, but I'm not sure how it ever worked.

In any case, I've fixed it up and added an example to UIExplorer. I also made a few changes to the API, but it's substantially the same.

**Comment by browniefed** [comment]
Thanks for cleaning it up, I did some hasty rebase conflicts with out a diff tool and screwed it up. Glad this got through though

**Pull request closed without merging**